src/browser.ts

```diff
@@ -74,17 +76,21 @@ export class GrpcClient {
       headers['Content-Type'] = 'application/x-protobuf';
       headers['User-Agent'] = 'testapp/1.0';
 
+      //const grpc_fallback_protocol = opts.protocol | 'https'
       const servicePath =
         opts.servicePath || method.parent.options['(google.api.default_host)'];
       const servicePort = opts.port || 443;
-      const serviceName = method.parent.parent.options.java_package.substring(
-        4
+      const serviceName = method.parent.parent.options.java_package.replace(
```

Comment:
use the proto package name instead? I think it'd be `method.parent.parent.package` and then use `method.parent.name` i.e. `google.cloud.vision.v1` + `ImageAnnotatorService`.

Reply:
Just checked, and it looks like `method.parent.parent.package` is undefined, at least for the methods as part of the Echo service. I'll continue to look into it, but so far haven't found any way to directly extract the service name from the method object... the only other way I can think of right now is to maybe extract parts of the name separately and put them together (i.e, like for `google.cloud.vision.v1`, extracting `google`, `cloud`, `vision` and `v1` separately and then putting them together) but that would probably mess up gapic-showcase usage

Reply:
@ramya-ramalingam I played a little bit with the service object. 

```
> service.name
'Echo'
> service.parent.name
'v1beta1'
> service.parent.parent.name
'showcase'
> service.parent.parent.parent.name
'google'
> service.parent.parent.parent.parent.name
''
```

So, it seems we can keep building an array `['google', 'showcase', 'v1beta1', 'Echo']` by moving from our `service` to its `.parent` and all the way up, and then `.join('.')` the resulting list to get the fully qualified name. @noahdietz from your experience, do you think it will work?

(by the way, `method.parent` is always `service` if I understand correctly, so some code around this place can be written shorter)

Reply:
Using `java_package` is not what we want because that option is there to make it possible to _override_ the package name in the generated Java code (should someone want to do it), so it's not guaranteed to be equal to the fully qualified package name.